plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,609 @@
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// converts `sentinel replicas <name>` output to tags and fields
+func convertSentinelReplicaOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	masterName string,
+	replica map[string]string,
+) (map[string]string, map[string]interface{}) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	tags["replica_ip"] = replica["ip"]
+	tags["replica_port"] = replica["port"]
+	tags["master"] = masterName
+
+	fields := make(map[string]interface{})
+
+	for key, val := range replica {
+		key = strings.Replace(key, "-", "_", -1)
+
+		switch valType := measurementReplicasFields[key]; valType {
+		case "float":
+			if val, err := strconv.ParseFloat(val, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "integer":
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// convertSentinelInfoOutput parses `INFO` command output
+// Largely copied from the Redis input plugin's gatherInfoOutput()
+func convertSentinelInfoOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	rdr io.Reader,
+) (map[string]string, map[string]interface{}) {
+	scanner := bufio.NewScanner(rdr)
+	fields := make(map[string]interface{})
+
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	var section string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+
+		if line[0] == '#' {
+			if len(line) > 2 {
+				section = line[2:]
+			}
+			continue
+		}
+
+		parts := strings.SplitN(line, ":", 2)
+		if len(parts) < 2 {
+			continue
+		}
+
+		name := parts[0]
+
+		if section == "Server" {
+			// Rename and convert to nanoseconds
+			if name == "uptime_in_seconds" {
+				if uptimeInSeconds, uptimeParseErr := strconv.ParseInt(parts[1], 10, 64); uptimeParseErr == nil {
+					fields["uptime_ns"] = int64(time.Duration(uptimeInSeconds) * time.Second)
+				} else {
+					acc.AddError(uptimeParseErr)
+				}
+
+				continue
+			}
+		} else if section == "Clients" {
+			// Rename in order to match the "redis" input plugin
+			if name == "connected_clients" {
+				name = "clients"
+			}
+		}
+
+		metric := strings.Replace(name, "-", "_", -1)
+
+		val := strings.TrimSpace(parts[1])
+
+		switch valType := measurementSentinelFields[metric]; valType {
+		case "float":
+			if val, err := strconv.ParseFloat(val, 64); err == nil {
+				fields[metric] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "integer":
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[metric] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[metric] = val
+		default:
+			continue
```

Comment:
This is unnecessary I think.

Reply:
@srebhan Addressed in https://github.com/influxdata/telegraf/commit/4c8fd93b34c4f9f6983d8300c2aa3bbb3ed25846